packages/flutter/lib/src/widgets/icon_data.dart

```diff
@@ -17,6 +21,9 @@ class IconData {
   ///
   /// The [fontPackage] argument must be non-null when using a font family that
   /// is included in a package. This is used when selecting the font.
+  ///
+  /// Instantiating non-const instances of this class can cause icon
+  /// tree-shaking to fail.
```

Comment:
Link this to something where people can learn more about icon tree shaking?

Reply:
Also, maybe document what failure means here: Will your app stop running/building? Is it that your app will function just fine, but the app size will be larger than necessary?

Reply:
ack about documenting the failiure mode.

AFAIK this is the first documentation we have about the icon tree-shaking feature. Should I link to the source?

Reply:
Maybe link it to staticIconProvider? That seems to be the canonical doc for what this tree shaking thing is for icons?

Reply:
Good idea.

Reply:
Done and done.